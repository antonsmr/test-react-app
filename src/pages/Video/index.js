/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { shape, func } from 'prop-types';
import { isEmpty } from 'lodash';
import { Row, Col } from 'antd';

import {
  getVideoList,
} from '../../actions/video';

import Layout from '../../components/Layout';

import { streamApiPath } from '../../utils/paths';

const VideoPage = (props) => {
  const [windowWidth, setWindowWidth] = useState(1000);
  const { video } = props;

  useEffect(() => {
    props.getVideoList();
  }, []);

  const updateDimensions = () => setWindowWidth(window.innerWidth);

  useEffect(() => {
    updateDimensions(window.innerWidth);
    window.addEventListener('resize', updateDimensions);

    return () => window.removeEventListener('resize', updateDimensions);
  }, []);

  const getGridSize = () => {
    let gridSize = 24;
    if (windowWidth >= 1200) gridSize = 8;
    if (windowWidth < 1200 && windowWidth >= 768) gridSize = 24;
    if (windowWidth <= 768) gridSize = 32;
    return gridSize;
  };

  return (
    <Layout>
      <h4>Video list</h4>

      <Row
        gutter={{
          xs: 8, sm: 16, md: 24, lg: 32,
        }}
      >
        {!isEmpty(video.list) && video.list.map((v) => (
          <Col key={v} className="gutter-row" span={getGridSize()}>
            <div className="video-list-item">
              <video id="video" controls="controls">
                <source src={`${streamApiPath}?fileName=${v}`} type="video/mp4" />
                <track src="captions_en.vtt" kind="captions" srcLang="en" label="english_captions" />
              </video>
            </div>
          </Col>
        ))}
      </Row>
    </Layout>
  );
};

VideoPage.propTypes = {
  video: shape().isRequired,
  getVideoList: func.isRequired,
};
VideoPage.defaultProps = {};

const mapStateToProps = (state) => ({
  video: state.video,
});

const actionCreators = {
  getVideoList,
};

export default connect(mapStateToProps, actionCreators)(VideoPage);
