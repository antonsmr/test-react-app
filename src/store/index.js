import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import combinedReducers from '../reducers';
import config from '../config';

export const configureStore = () => {
  const logger = createLogger({
    collapsed: (getState, action, logEntry) => !logEntry.error,
  });

  const create = () => {
    if (config.env === 'development') {
      return createStore(combinedReducers, composeWithDevTools(
        applyMiddleware(
          thunk,
          logger,
        ),
      ));
    }

    return createStore(combinedReducers, applyMiddleware(thunk));
  };

  const store = create();

  if (module.hot) {
    module.hot.accept('../reducers', () => {
      store.replaceReducer(combinedReducers);
    });
  }

  return store;
};

export default configureStore;
