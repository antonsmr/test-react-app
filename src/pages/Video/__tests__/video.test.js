import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Video from '../index';
import { configureStore } from '../../../store';

import '@testing-library/jest-dom';

configure({ adapter: new Adapter() });

const mockProps = {
  video: {
    fetching: false,
    fetched: false,
    list: [],
  },
  getVideoList: jest.fn(),
};

const store = configureStore();

describe('Video page', () => {
  it('should be defined', () => {
    expect(Video).toBeDefined();
  });

  test('should run example code', () => {
    expect(mockProps.video).toBeDefined;
    expect(mockProps.getVideoList).toBeDefined;
    expect(mockProps.getVideoList).toMatchSnapshot();
  });

  it('shallow render test', () => {
    shallow(
      <Router>
        <Provider store={store}>
          <Video {...mockProps} />
        </Provider>
      </Router>,
    );
  });
});
