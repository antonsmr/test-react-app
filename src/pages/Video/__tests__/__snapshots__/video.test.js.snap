// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video page should run example code 1`] = `[MockFunction]`;
